mne/decoding/transformer.py

```diff
@@ -520,3 +521,94 @@ def transform(self, epochs_data, y=None):
                                      picks=self.picks, n_jobs=self.n_jobs,
                                      copy=False, verbose=False)
         return epochs_data
+
+
+class UnsupervisedSpatialFilter(TransformerMixin, BaseEstimator):
+    """Fit and transform with an unsupervised spatial filtering across time
+    and samples.
+
+    Parameters
+    ----------
+    estimator : scikit-learn estimator
+        Estimator using some decomposition algorithm.
+    average : bool, defaults to False
+        If True, the estimator is fitted on the average across samples
+        (e.g. epochs).
+    """
+    def __init__(self, estimator, average=False):
+        self._check_estimator(estimator, average)
+        self.estimator = estimator
+        self.average = average
+
+    def _check_estimator(self, estimator, average):
```

Comment:
I meant external to the object. If #3381 is merged, used the `_check_estimator` from there. Else leave an XXX, and I'll make a PR to minimize the code redundancy later.